### PR TITLE
tensorflow-lite: Add v2.15.0

### PR DESCRIPTION
**Pull request by talyz**
Specify library name and version:  **tensorflow-lite/v2.15.0**

Add 2.15.0 and update a few dependencies which were causing crashes when running on M1 macs under rosetta. Also, remove 2.12.0, which was just a copy of 2.10.0.

Depends on #21762, #21761, #21760 and #21759, which should be merged before this.

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by minus7** [comment]
Removal of 2.12.0 closes #20170

**Review by jwillikers — APPROVED** [review]

**Comment by talyz** [comment]
Rebased on master, now that the dependent package updates are merged.

**Review by jwillikers — APPROVED** [review]

**Review by jwillikers — APPROVED** [review]

**Comment by mologie** [comment]
Likewise using this on macOS, iOS, and Linux without modification. Cross-compiling to Android requires working around a [problem in xnnpack/clang](https://github.com/google/XNNPACK/issues/4348), but that's an upstream issue and I'll provide a separate PR to the xnnpack recipe once this one is merged. LGTM, thank you @talyz for building this integration!

(I am not associated with JFrog, still needs maintainer approval.)

**Comment by talyz** [comment]
Just rebased on master to get the separate dependency update in. @uilianries is there anything more that needs to happen for this to be reviewed / merged?

File: recipes/tensorflow-lite/all/patches/dependencies_2_15.patch
Comment (valgur) [inline recipes/tensorflow-lite/all/patches/dependencies_2_15.patch:1]: Can be simplified or even avoided by using `CMakeDeps.set_property()` on the `cmake_file_name` and `cmake_target_name` properties.

**Review by valgur — APPROVED** [review]

**Comment by mologie** [comment]
Authentic Vision is now using this recipe as-is in production since April with no known issues on Ubuntu 22.04, 24.04, iOS 13-17, and Android 6-14. The XNNPACK change at #22951 was recently merged to allow building for Android -- if anyone following this PR wants to do that, bump your caches :).

We'd eventually want to bump TensorFlow and its dependencies to 2.15.1 and 2.16.X, would appreciate if this could be reviewed again and merged in preparation for that.

**Comment by RubenRBS** [comment]
> We'd eventually want to bump TensorFlow and its dependencies to 2.15.1 and 2.16.X, would appreciate if this could be reviewed again and merged in preparation for that.

We'll push this review forward this next week, thanks a lot for taking the time to create the PR in the first place. Could you fix the conandata conflicts meanwhile? Thanks!

**Comment by mologie** [comment]
> We'll push this review forward this next week, thanks a lot for taking the time to create the PR in the first place. Could you fix the conandata conflicts meanwhile? Thanks!

Thanks for the update @RubenRBS, the original author is @talyz though and probably received your comment with this thread anyway. I have just been using this recipe from this branch and worked on Android dependencies a bit.

**Comment by mologie** [comment]
The bot is right @talyz, `patches/disable_fetch_content.patch` referenced in conandata.yml does not exist in the source tree. (It's unclear to me why this does not fail with a FileNotFoundError at some point, did not dig into Conan for that.)

Other patches have a version prefix and order in their filenames, would thus suggest to match that style for consistency.

Once fixed, this should pass CI.

**Review by czoido — APPROVED** [review]

**Review by AbrilRBS — APPROVED** [review]
10/10 :)

**Pull request merged**
